cpp/base/notification.cc

```diff
@@ -9,16 +9,20 @@ using std::unique_lock;
 
 namespace cert_trans {
 
+Notification::~Notification() {
+  // Ensure nobody is inside Notify() when we're destructing.
```

Comment:
It's definitely not fixing *this* problem, but ok, I'll take it out - just understand that the next time there's a 1 in 10,000 spurious hang, you've already volunteered to find it :)